streams/src/test/java/org/apache/kafka/streams/state/internals/metrics/RocksDBMetricsRecorderTest.java

```diff
@@ -279,449 +310,335 @@ public void shouldThrowIfDbToAddWasAlreadyAddedForOtherSegment() {
-        numberOfOpenFilesSensor.record((5 + 7) - (3 + 4), 0L);
-        replay(numberOfOpenFilesSensor);
-
-        expect(statisticsToAdd1.getAndResetTickerCount(TickerType.NO_FILE_ERRORS)).andReturn(34L);
-        expect(statisticsToAdd2.getAndResetTickerCount(TickerType.NO_FILE_ERRORS)).andReturn(11L);
-        numberOfFileErrorsSensor.record(11 + 34, 0L);
-        replay(numberOfFileErrorsSensor);
-
-        replay(statisticsToAdd1);
-        replay(statisticsToAdd2);
-
-        recorder.record(0L);
-
-        verify(statisticsToAdd1);
-        verify(statisticsToAdd2);
-        verify(
-            bytesWrittenToDatabaseSensor,
-            bytesReadFromDatabaseSensor,
-            memtableBytesFlushedSensor,
-            memtableHitRatioSensor,
-            memtableAvgFlushTimeSensor,
-            memtableMinFlushTimeSensor,
-            memtableMaxFlushTimeSensor,
-            writeStallDurationSensor,
-            blockCacheDataHitRatioSensor,
-            blockCacheIndexHitRatioSensor,
-            blockCacheFilterHitRatioSensor,
-            bytesWrittenDuringCompactionSensor,
-            bytesReadDuringCompactionSensor,
-            compactionTimeAvgSensor,
-            compactionTimeMinSensor,
-            compactionTimeMaxSensor,
-            numberOfOpenFilesSensor,
-            numberOfFileErrorsSensor
-        );
+        when(statisticsToAdd1.getHistogramData(HistogramType.COMPACTION_TIME)).thenReturn(compactionTimeData1);
+        when(statisticsToAdd2.getHistogramData(HistogramType.COMPACTION_TIME)).thenReturn(compactionTimeData2);
+        doNothing().when(compactionTimeAvgSensor).record((double) (8 + 8) / (2 + 2), now);
+        doNothing().when(compactionTimeMinSensor).record(4.0, now);
+        doNothing().when(compactionTimeMaxSensor).record(24.0, now);
+
+        when(statisticsToAdd1.getAndResetTickerCount(TickerType.NO_FILE_OPENS)).thenReturn(5L);
+        when(statisticsToAdd1.getAndResetTickerCount(TickerType.NO_FILE_CLOSES)).thenReturn(3L);
+        when(statisticsToAdd2.getAndResetTickerCount(TickerType.NO_FILE_OPENS)).thenReturn(7L);
+        when(statisticsToAdd2.getAndResetTickerCount(TickerType.NO_FILE_CLOSES)).thenReturn(4L);
+        doNothing().when(numberOfOpenFilesSensor).record((5 + 7) - (3 + 4), now);
+
+        when(statisticsToAdd1.getAndResetTickerCount(TickerType.NO_FILE_ERRORS)).thenReturn(34L);
+        when(statisticsToAdd2.getAndResetTickerCount(TickerType.NO_FILE_ERRORS)).thenReturn(11L);
+        doNothing().when(numberOfFileErrorsSensor).record(11 + 34, now);
+
+        recorder.record(now);
+
+        verify(statisticsToAdd1, times(17)).getAndResetTickerCount(isA(TickerType.class));
+        verify(statisticsToAdd2, times(17)).getAndResetTickerCount(isA(TickerType.class));
+        verify(statisticsToAdd2, times(2)).getHistogramData(isA(HistogramType.class));
+        verify(statisticsToAdd2, times(2)).getHistogramData(isA(HistogramType.class));
```

Comment:
You need to verify each `TickerType` and each `HistogramType` here. Using `isA()` is to general. I think the best thing would be to split this test method in one test method per computed metric.

Reply:
@cadonna, @divijvaidya, what do you think we just update this test method with "improved" verification, as @cadonna described [here](https://github.com/apache/kafka/pull/12459/files/e4e7549e229b2687f8491842a2352ecaf2563e5f#r944335669), and we create a new ticket/PR to split this test to multiple smaller tests (which would be cool)?

IMHO, that way we keep this PR short and simple and in the scope of the "Replace EasyMock and PowerMock with Mockito.." task 🤔

Reply:
I am fine with both. However, I am not sure whether the "improved" verification vs. splitting unit tests will make too much difference.

Reply:
Regarding effort-wise, it should be the same. 

The only deficiency in splitting unit tests in this PR, whose main purpose is to replace `EasyMock` is that it makes it  a little bit harder to review (since we're introducing changes that are out-of-the scope).

Reply:
I think I agree with you!

Reply:
Thanks for the review and discussion here folks. I have tried to optimize this PR for code review friendliness and I agree that we should probably leave refactoring and other changes out of scope. I have addressed your comments Bruno. Please take a look when you get a chance.